imap_processing/cdf/utils.py

```diff
@@ -41,6 +42,24 @@ def calc_start_time(shcoarse_time: float) -> np.datetime64:
     return launch_time + time_delta
 
 
+def load_cdf(file_path: Path, **kwargs: dict) -> xr.Dataset:
+    """Load the contents of a CDF file into an ``xarray`` dataset.
+
+    Parameters
+    ----------
+    file_path : Path
+        The path to the CDF file
+    **kwargs : dict, optional
+        Keyword arguments for ``cdf_to_xarray``
+
+    Returns
+    -------
+    xr.Dataset
+        The ``xarray`` dataset for the CDF file
+    """
+    return cdf_to_xarray(file_path, kwargs)
```

Comment:
I may have found a bug with this `cdf_to_xarray`. When we read cdf file using this, it returns global attrs as dictionary but all the values are returned as list which is ok if we are only reading the cdf and not using it for subsequent data level. But I want to carry l1a attrs for l1a with minor updates. I am sure others may want to use similar manner. If so, I wonder if we can extend this function to convert global attrs values back into strings instead of list here or update on cdflib. @bryan-harter may know which way is better. This is a comment. We don’t have to add this in this PR. We can extend this in upcoming PR.

Reply:
Glad you added this common functions!

Reply:
Yeah, I also noticed this, which also makes it a bit difficult to test. Good idea to put the fix into this function, though I wonder if the fix should be made 'upstream' in `cdflib`? Interested to hear @bryan-harter thoughts on this.

Reply:
So by default the cdf reader function just assumes all attributes are lists because they technically are all lists from the CDF's point of view. When writing an attribute, you specify how long the list is you are about to write. So there is technically no way to tell if an attribute is simply a single item v.s. a list with a single item in it.  

One thing I could do is have cdflib assume all lists with only 1 item are returned as not lists.  But just be aware that means that if you DO feed in the attribute as a list with one item, this code change would cause the opposite problem and get rid of the list.  

I'm open to ideas!

Reply:
Thanks @bryan-harter!

My gut says that it makes sense to deal with the single-item vs list logic within `write_cdf()`/`load_cdf()` and keep `cdflib` to be "closer" to the CDF requirements / more generalized so that issues doesn't arise for other developers 🤷🏻

Reply:
Thank you guys for already tackling this in this PR!